### PR TITLE
travis: upgrade and cache pip

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

**Comment by jaraco** [comment]
What's the value in upgrading pip if all it installs is tox?

**Comment by xoviat** [comment]
I was thinking that we may want integration tests for PEP 517.

File: .travis.yml [hunk 1]
```diff
@@ -8,13 +8,19 @@ python:
   - 3.6
   - nightly
   - pypy
+cache:
```
Comment (webknjaz) [inline .travis.yml]: Just `cache: pip` would work. And also there's this syntax:
```yaml
cache:
  pip: true
```

I don't really like the idea of hardcoding paths.

Comment (webknjaz) [inline .travis.yml]: (ref: https://github.com/pypa/setuptools/pull/1053/files?diff=split#diff-354f30a63fb0907d4ad57269548329e3R34)

**Comment by webknjaz** [comment]
Is this needed for all envs?

**Comment by jaraco** [comment]
This PR needs a better explanation of its motivations and value. I can infer that enabling pip cache is a good thing, but if it's a best practice, then we should be able to reference documentation or a conversation that indicates that this setting is a best practice (and if it is, I'll want to incorporate it with my other projects).

As for upgrading pip, that feels like a separate change with separate motivations that aren't even borne out. I don't know PEP 517 by name, but even if I did, it's not clear to me why upgrading pip in this commit helps. There should be a separate pull request that summarizes what PEP 517 does and how Setuptools can better support that and including the integration tests that go along with the upgrading pip stage.

**Pull request closed without merging**